### PR TITLE
PR for #111: drop of collinear variables

### LINKED ISSUE #111: drop of collinear variables in the IV's check of collinearity
When _eventiv checks for collinearity between the IV and the event-time dummies, it does so by regressing the IV on the dummies and checking if r2 is 1. This is not a good way of doing this, because Stata will drop collinear variables. We should be doing this with _rmcoll.

Derived from #109 https://github.com/JMSLab/xtevent/issues/109#issuecomment-1261608505

### PR DESCRIPTION
**Pull request by Constantino-Carreto-Romero**
Closes #111

### CODE REVIEW

**Comment by Constantino-Carreto-Romero** [comment]
Solution in https://github.com/JMSLab/xtevent/pull/112/commits/96553b715868f59a0decde30d74509c20889e288 is incomplete. There can be cases when the dependent variable is not collinear, but there is collinearity among the independent variables. In that case, the error message will appear, and it shouldn't.

**Comment by Constantino-Carreto-Romero** [comment]
In https://github.com/JMSLab/xtevent/pull/112/commits/44e0bb2e50d51f863ebc6f85c321bda547f75ad3 the collinearity check is done pairwise, one instrument against one event-time dummy. For instance, for the 3rd example in the homepage:
```
*setup
webuse nlswork, clear
xtset idcode year

*FHS estimator with proxy variables
xtevent ln_w age c.age#c.age ttl_exp c.ttl_exp#c.ttl_exp tenure , ///
            pol(union) w(3) vce(cluster idcode) impute(nuchange) ///
            proxy(wks_work)
```

Previously, executing it produced this [outcome](https://user-images.githubusercontent.com/36095845/193622244-349b063c-6790-4d1f-8c92-e590552cb6e4.png), while it now produces the following [error message](https://user-images.githubusercontent.com/36095845/193622374-7bc2c845-101e-4914-b8a3-25e251f5bf27.png).

Previous examples in https://github.com/JMSLab/xtevent/issues/111#issuecomment-1264052808 keep working as intended.

File: xtevent/_eventiv.ado [hunk 1]
```diff
@@ -234,13 +234,17 @@ program define _eventiv, rclass
...
-
+	
 	foreach v in `leadivs' `varivs' {
-		qui _regress `v' `included' [`weight'`exp'] if `touse', absorb(`i')
-		if e(r2)==1 {
-			di as err "Instrument is collinear with the included event-time dummies. You may have generated leads of the policy variable and included them in the proxyiv option instead of specifying the lead numbers."
-			exit 301
-		}	
+		foreach evtd in `included'{
+			qui _rmcoll `v' `evtd' if `touse'
```
Comment (jorpppp) [inline xtevent/_eventiv.ado:240]: @Constantino-Carreto-Romero This is checking collinearity of each iv against a single event-time dummy at each time, but we want to check collinearity with all the event time dummies at the same time. So I think here you need something like 
```_rmcoll `v' `included`if `touse' ```

If there's collinearity and you still want the error message to single out the collinear variable, you can probably loop over the combinations of incluided using `tuples`.

**Comment by Constantino-Carreto-Romero** [comment]
@jorpppp in https://github.com/JMSLab/xtevent/pull/112/commits/1fa4baed2e6b013f2e0994f6cbec54412df0c97a I'm using `_rmdcoll` instead of `_rmcoll`. By using `_rmdcoll`, it only focuses on checking whether the independent variable (the instrument) is collinear with any combination of the dependent variables (the event-time dummies). If so, it produces an error. 
These are some examples to the implementation:
```
*install the branch version 
net install xtevent, from("https://raw.githubusercontent.com/JMSLab/xtevent/issue111_drop_of_collinear_variables") replace

*load example dataset
use "https://github.com/JMSLab/xtevent/blob/main/test/example31.dta?raw=true", clear
xtset i t

*generate collinear variables
gen lead2=f2.d.z
gen lead3_lead2 = f3.d.z + f2.d.z

*expect an error 
xtevent y x, panelvar(i) t(t) policyvar(z) window(3) proxy(zeta) proxyiv(lead2) 
xtevent y x, panelvar(i) t(t) policyvar(z) window(3) proxy(zeta) proxyiv(lead3_lead2) 

*no error message 
xtevent y x, panelvar(i) t(t) policyvar(z) window(3) proxy(zeta) proxyiv(1 2) 
xtevent y x, panelvar(i) t(t) policyvar(z) window(3) proxy(zeta) proxyiv(2) 
xtevent y x, panelvar(i) t(t) policyvar(z) window(3) proxy(zeta)

```

**Comment by jorpppp** [comment]
@Constantino-Carreto-Romero Thanks, this loosk good. Merging now.

**Pull request merged**